src/framework/theme/components/stepper/stepper.component.ts

```diff
@@ -147,14 +147,35 @@ export class NbStepperComponent {
    * Navigate to next step
    * */
   next() {
-    this.selectedIndex = Math.min(this.index + 1, this.steps.length - 1);
+    const nextStepIndex = Math.min(this.selectedIndex + 1, this.steps.length - 1);
+    const selectedStep = this.steps.toArray()[this.selectedIndex];
+    if (selectedStep.stepControl) {
```

Comment:
I think we can remove `if` nesting here and just write (selectedStep.stepControl && selectedStep.isValid).
Moreover, to me It would be better to move this condition in the separate function.

Reply:
Furthermore this if\else navigation has to be moved in the separate function because it's used not only here.